prow/logrusutil/logrusutil.go

```diff
@@ -27,39 +27,49 @@ import (
 type DefaultFieldsFormatter struct {
 	WrappedFormatter logrus.Formatter
 	DefaultFields    logrus.Fields
+	PrintLineNumber  bool
 }
 
-// NewDefaultFieldsFormatter returns a DefaultFieldsFormatter,
-// if wrappedFormatter is nil &logrus.JSONFormatter{} will be used instead
-func NewDefaultFieldsFormatter(
-	wrappedFormatter logrus.Formatter, defaultFields logrus.Fields,
-) *DefaultFieldsFormatter {
-	res := &DefaultFieldsFormatter{
-		WrappedFormatter: wrappedFormatter,
-		DefaultFields:    defaultFields,
+// Init set Logrus formatter
+// if DefaultFieldsFormatter.wrappedFormatter is nil &logrus.JSONFormatter{} will be used instead
+func Init(formatter *DefaultFieldsFormatter) {
+	if formatter == nil {
+		return
 	}
-	if res.WrappedFormatter == nil {
-		res.WrappedFormatter = &logrus.JSONFormatter{}
+	if formatter.WrappedFormatter == nil {
+		formatter.WrappedFormatter = &logrus.JSONFormatter{}
 	}
-	return res
+	logrus.SetFormatter(formatter)
+	logrus.SetReportCaller(formatter.PrintLineNumber)
+}
+
+// ComponentInit is a syntax sugar for easier Init
+func ComponentInit(component string) {
+	Init(
+		&DefaultFieldsFormatter{
+			PrintLineNumber: true,
```

Comment:
You might want to define a `CallerPrettyfier` to trim GOPATH. Otherwise, paths will be prefixed with something like `/Users/username/go/k8s.io/test-infra/prow/...`.